### ISSUE TITLE
How can I extract uniprot ids and corresponding foldseek sequences from a pre-generated database?

### ISSUE THREAD

**Issue by LTEnjoy** [steineggerlab/foldseek#201]
Hi!

Thank you for your great work! I have a question that whether I can download a pre-generated database and manually generate a fasta file containing all protein names and corresponding foldseek sequences.

For example for  the ``alphafold_swissprot`` database, I want to extract from this database all UniProt IDs and foldseek sequences and write it into a fasta file like:

> uniprot_id_1
xxxxxxxxxxxxxxxxxxx

> uniprot_id_2
xxxxxxxxxxxxxxxxxxxxxxxxxx

Thank you in advance and I'm looking forward to your reply!

**Comment by milot-mirdita** [comment]
You can use `createsubdb` with a list of accessions and then call `covert2fasta` to make a FASTA file:

```
foldseek createsubdb accession_list alphafold_swissport afsp_subset --id-mode 1
foldseek convert2fasta afsp_subset afsp_subset.fasta
```

Please check that the accessions you pass are in the same format as the ones that are stored in the second column of the `alphafold_swissport.lookup` file.

**Comment by LTEnjoy** [comment]
Thank for your quick reply! I tried above commands and it indeed generated a fasta file!

It's just slightly different than what I thought as I want to get sequences encoded by foldseek, not the residue sequences. Could you tell me how to generate that kind of fasta file?

Thank you again!

**Comment by LTEnjoy** [comment]
That's exactly what I want!

Thank you very much! Have a nice day!

**Comment by LTEnjoy** [comment]
Hello,

When I tried the command ``foldseek createsubdb accession_list alphafold_swissport_ss afsp_subset_ss --id-mode 1`` on af50db, I got these errors:

<img width="422" alt="1698823252767" src="https://github.com/steineggerlab/foldseek/assets/52776915/c9006799-48a4-4a7e-b681-fc68d5334d5b">

Could you tell how I can fix this problem? I want to generate all UniProt 3Di sequences from this database.

**Comment by milot-mirdita** [comment]
I think you have to run first:

```
ln -s alphafold_swissport.lookup alphafold_swissport_ss.lookup
```

**Comment by LTEnjoy** [comment]
You are right. In my case, I just directly substituted the ``xx_ss.lookup`` with ``xx.lookup``. I don't know the implementation but I guess this caused some mismatches.

**Comment by LTEnjoy** [comment]
I just tried this command, but errors still exist.
![image](https://github.com/steineggerlab/foldseek/assets/52776915/1595523e-4157-4f31-bf3e-f7a2bc948ffd)

Also, here are some contents in my ``accession_list.txt``:
![image](https://github.com/steineggerlab/foldseek/assets/52776915/2402caec-a512-414b-b7f7-0b4009b773b6)

**Comment by milot-mirdita** [comment]
Could you please post all commands you executed (preferably as text and not as screenshots)? I am not sure what's going on currently.

**Comment by LTEnjoy** [comment]
Hi,

I guess I found what the problem was. the ``afdb50`` only contains 50M sequences after clustering. But what I need is to generate sequences from the whole UniProt database (with ~200M sequences). So I downloaded the ``afdb`` database, by which I think the problem should be solved.
